packages/calypso-e2e/src/hooks.ts

```diff
@@ -64,24 +64,26 @@ export const setupHooks = ( callback: ( { page }: { page: Page } ) => void ): vo
 			await page.screenshot( { path: fileName } );
 		}
 		// Close the browser. This needs to be called before trying to access
-		// the video recording
+		// the video recording.
 		await page.close();
-		// Save video
+
 		if ( __CURRENT_TEST_FAILED__ ) {
-			const original = await ( page.video() as Video ).path();
+			// const original = await ( page.video() as Video ).path();
 			const destination = path.join(
 				artifactPath,
 				'screenshots',
 				`${ getTestNameWithTime( testName ) }.webm`
 			);
 			try {
-				await rename( original, destination );
+				// Save the failing test case with a specific name.
+				await page.video()?.saveAs( destination );
```

Comment:
Hrm, do we need to make this work for new contexts? Looking at the code, it didn't work for it before (because the path was still based on the current `page`s video.

I'm okay with punting on that for now to a future PR if you are! That might be kind of tricky to solve...

Reply:
This will not work for any context other than the instance of `page` instantiated earlier in the hook. This is indeed an issue that's been around for some time and reason why contexts launched in the specs did not have video recording captured even for failing test cases.

To work around this issue, we need to have a way to keep track of all contexts opened for the test spec.
`page.contexts` will return `Array<BrowserContext>` and so one possible approach could be that for each BrowserContext in the Array:

1. obtain list of pages open in the BrowserContext.
2. close the pages.
3. give it a unique name (somehow).
4. move them to the artifact dir.

Repeat for every BrowserContext that is open, and then close the browser instance once complete.

Reply:
Filed https://github.com/Automattic/wp-calypso/issues/56805.